be/src/storage/vectorized/column_predicate_rewriter.cpp

```diff
@@ -160,11 +161,12 @@ bool ColumnPredicateRewriter::_rewrite_predicate(ObjectPool* pool, const FieldPt
         }
         if (PredicateType::kLE == pred->type() || PredicateType::kLT == pred->type()) {
             _get_segment_dict(&sorted_dicts, _column_iterators[cid]);
-            std::vector<std::string> str_codewords;
-            auto value = pred->value().get_slice().to_string();
+            // use non-padding string value.
+            auto value = pred->values()[0].get_slice().to_string();
```

Comment:
I think we should add some comments to `value()` and `values()` method in `ColumnPredicate`, let other people know the char type has the padding issue,  the `value()` and `values()`  are different

Reply:
OK.